app/src/main/cpp/skyline/services/am/storage/IStorageAccessor.cpp

```diff
@@ -8,34 +8,38 @@ namespace skyline::service::am {
     IStorageAccessor::IStorageAccessor(const DeviceState &state, ServiceManager &manager, std::shared_ptr<IStorage> parent) : parent(std::move(parent)), BaseService(state, manager) {}
 
     Result IStorageAccessor::GetSize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        response.Push<i64>(static_cast<i64>(parent->content.size()));
+        response.Push<i64>(static_cast<i64>(parent->GetSpan().size()));
         return {};
     }
 
     Result IStorageAccessor::Write(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
         auto offset{request.Pop<i64>()};
 
-        if (offset < 0 || offset > parent->content.size())
+        if (!parent->writable)
+            return result::ObjectInvalid;
+        span<u8> storageSpan{parent->GetSpan()};
```

Comment:
You don't necessarily need to update this but for future reference, in cases like this we tend to use `auto` rather than spelling out a templated type.